### ISSUE TITLE
Different behavior of model.predict and .call if training=True at dropout layer with pre-set seed

### ISSUE THREAD

**Issue by marcohuber** [tensorflow/tensorflow#44614]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Win10
- TensorFlow version (use command below): v2.3.0-54-gfcc4b966f1 2.3.1
- Python version: 3.7

**Describe the current behavior**
Different behavior of `model.predict `and `.call (model(...)),` if `training=True` is used at dropout layer. It seems that the call method does not take into account pre-set dropout seeds, while the predict method does.

**Describe the expected behavior**
Same behavior of the predict and call method if the same dropout seed is set and `training=True`.

**Standalone code to reproduce the issue**
```
import numpy as np
import tensorflow as tf

from tensorflow.keras import layers
from tensorflow.keras.models import Model

def build_toy():
    
    inputs = layers.Input(shape=(1,), name="inputs")
    x = layers.Dense(1, activation="linear", kernel_initializer=tf.keras.initializers.Constant(value=1), name="dense") (inputs)
    x = layers.Dropout(0.5, seed=12, name="dropout") (x,  training=True)
    out = x
    
    model = Model(inputs, out)
    
    return model

data = np.array(([1]))
    
predict_model = build_toy()
print(predict_model.predict(data))

call_model = build_toy()
print(call_model(data))

```
`predict_model` always predicts the same value every time the model is newly built, while `call_model(data) ` may changes.

**Comment by amahendrakar** [comment]
Was able to reproduce the issue with TF v2.3 and TF-nightly. Please find the gist of it [here](https://colab.research.google.com/gist/amahendrakar/e752bda68103226a37c72fce2e2379bc/44614.ipynb). Thanks!

**Comment by sachinprasadhs** [comment]
Was able to reproduce your issue in Tf Nightly 2.6.0-dev20210530, please find the gist [here](https://colab.research.google.com/gist/sachinprasadhs/d9725afc44102625ff5c686ee8603dff/44614.ipynb). Thanks!

**Comment by mohantym** [comment]
Hi @marcohuber! I was able to resolve this issue by randomizing seed value . Attaching Gist in TF [2.7 ](https://colab.research.google.com/gist/mohantym/f6226fe79ea3a613ca5c794044a812aa/44614.ipynb#scrollTo=UMHAOybY7C5F)and [nightly](https://colab.research.google.com/gist/mohantym/17f98a6570db540c754553506844fd70/44614.ipynb#scrollTo=ITWRA5fO7Hco) for reference. Thank you!

**Comment by marcohuber** [comment]
Hi @mohantym, Thanks for the fix!
